frontend/src/lib/k8s/networkpolicy.tsx

```diff
@@ -0,0 +1,41 @@
+import { apiFactoryWithNamespace } from './apiProxy';
+import { KubeObjectInterface, LabelSelector, makeKubeObject } from './cluster';
+
+interface NetworkPolicyPort {
+  port: string | number;
+  protocol: string;
+}
+
+interface IPBlock {
+  cidr: string;
+  except: string;
+}
+
+interface NetworkPolicyPeer {
+  ipBlock: IPBlock;
+  namespaceSelector: LabelSelector;
+  podSelector: LabelSelector;
```

Comment:
The optionality of these fields shouldn't be checked, I think some may be optional.